packages/commonwealth/client/scripts/state/api/threads/deleteThread.ts

```diff
@@ -0,0 +1,52 @@
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+import { ThreadStage } from 'models/types';
+import app from 'state';
+import { EXCEPTION_CASE_threadCountersStore } from '../../ui/thread';
+import { removeThreadFromAllCaches } from './helpers/cache';
+
+interface DeleteThreadProps {
+  chainId: string;
+  threadId: number;
+}
+
+const deleteThread = async ({ chainId, threadId }: DeleteThreadProps) => {
+  return await axios.delete(`${app.serverUrl()}/threads/${threadId}`, {
+    data: {
+      jwt: app.user.jwt,
+      chain_id: chainId,
+    },
+  });
+};
+
+interface UseDeleteThreadMutationProps {
+  chainId: string;
+  threadId: number;
+  currentStage: ThreadStage;
+}
+
+const useDeleteThreadMutation = ({
+  chainId,
+  threadId,
+  currentStage,
+}: UseDeleteThreadMutationProps) => {
+  return useMutation({
+    mutationFn: deleteThread,
+    onSuccess: async (response) => {
+      // Update community level thread counters variables
+      EXCEPTION_CASE_threadCountersStore.setState(
```

Comment:
Thinking about it -- the way around the thread counters issue could be to augment the react-query config for topics, so that each topic knows how many threads it has. Something that might be worth ticketing.

Reply:
> the way around the thread counters issue could be to augment the react-query config for topics, so that each topic knows how many threads it has

this seems like a better idea. But we would also have to store the total number of threads in voting stage somewhere I assume (we already do it in code).

Other than that, each `/threads?bulk=true` will be requiring total available threads count for the provided query parameters.

Reply:
Please put up a ticket and we can mark this resolved.

Reply:
I think we already have one here https://github.com/hicommonwealth/commonwealth/issues/3244, no?

Reply:
or we could update that to include this case?

Reply:
Feel free to add a comment to #3244 linking back here + explaining the considerations and that should suffice.